server/controllers/profile.js

```diff
@@ -0,0 +1,102 @@
+const Profile = require("../models/Profile");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /profile/create
+// @desc create a profile
+// @access Private
+exports.createProfile = asyncHandler(async (req, res, next) => {
+    const userId = req.user.id;
+
+    const profileExists = await Profile.findOne({ userId:userId });
+    if (profileExists) {
+        res.status(400);
+        throw new Error("A user profile already registered");
+    }
+
+    const {
+        firstName, lastName, description, gender,
+        address, birthDate, phoneNumber, isAvailable, availability
+    } = req.body
+    
+    const profile = await Profile.create({
+        userId,
+        firstName,
+        lastName,
+        description,
+        gender,
+        birthDate,
+        phoneNumber,
+        address,
+        isAvailable,
+        availability,
+    })
+
+    if(!profile) {
+        res.status(400);
+        throw new Error("Invalid request data");
+    }
```

Comment:
Would this ever be hit? We should be doing middleware validation on the incoming request data to ensure reliability and keep our controller logic lean.